File: app/src/main/java/sgtmelon/scriptum/office/st/DragListenerSt.java
package sgtmelon.scriptum.office.st;

import sgtmelon.scriptum.office.intf.ItemIntf;

/**
 * Состояние для перетаскивания элементов, определяющее можно перетаскивать его или нет
 */
public final class DragListenerSt implements ItemIntf.DragListener {

    private boolean drag = false;

    public boolean isDrag() {
        return drag;
    }

    @Override
    public void setDrag(boolean drag) {
        this.drag = drag;
    }

}

File: app/src/main/java/sgtmelon/scriptum/office/st/DragListenerSt.kt
package sgtmelon.scriptum.office.st

import sgtmelon.scriptum.office.intf.ItemIntf

/**
 * Состояние для перетаскивания элементов, определяющее можно перетаскивать его или нет
 */
class DragListenerSt : ItemIntf.DragListener {

    private var drag = false

    fun isDrag(): Boolean {
        return drag
    }

    override fun setDrag(drag: Boolean) {
        this.drag = drag
    }

}